corehq/ex-submodules/casexml/apps/phone/tasks.py

```diff
@@ -22,3 +28,27 @@ def force_update_cleanliness_flags():
     # that there are no bugs in the weekly task.
     # If we haven't seen any issues by the end of 2015 (so 6 runs) we should remove this.
     set_cleanliness_flags_for_all_domains(force_full=True)
+
+
+@task(queue=ASYNC_RESTORE_QUEUE)
+def get_async_restore_payload(restore_config):
+    """Process an async restore
+    """
+    response = restore_config.generate_payload(async_task=current_task)
+
+    # delete the task id from the task, since the payload can now be fetched from the cache
+    restore_config.cache.delete(restore_config.async_cache_key)
+
+    return response
+
+
+@after_task_publish.connect
+def update_celery_state(sender=None, body=None, **kwargs):
+    """Used to update celery state so we know if an async restore made it to celery
```

Comment:
Good question. This sets the status of any `phone` task that gets sent to celery to "SENT".
When fetching a task from celery with `AsyncResult(task_id)`, if the task with `task_id` never existed, celery doesn't give you an error, instead it will just wait forever for the task to arrive, and to complete when trying to fetch the result. I think this would basically gridlock the phone waiting for an async restore (until the task_id cache expires)
Confusingly, the status for a task that never existed, and a task that was actually sent to celery is `PENDING`. What this does is updates the status for a task that does exist to `SENT`, so that we know whether it is in the queue.
I think this would only happen if celery gets restarted while processing an async restore, but after realizing it was a possibility, was really scared that it might happen. 
I think I would be easily convinced that I overthought this.